### PR TITLE
Added automatic override of dispatch timeout of UVM when sandbox receives a timeout option.

### PR DESCRIPTION
**Pull request by shamasis**
This fixes issue where `error` event was not raised when script has timed out.

### CODE REVIEW

**Comment by czardoz** [comment]
@shamasis Builds are failing on Node v6

**Comment by shamasis** [comment]
@czardoz - the tests were failing because I wrote the tests wrong. Perils of coding on evented systems! Some error events were fired first (based on whether the VM script runs fast or the outer script.) Seems like on Node v6, VM is faster. :-)

File: lib/index.js [hunk 1]
```diff
@@ -64,30 +69,41 @@ _.assign(PostmanSandbox.prototype, {
...
     },
 
     dispose: function () {
-        Object.keys(this.executing).forEach(function (id) {
-            this.emit(`execution.${id}`, new Error('sandbox: execution interrupted, bridge disconnecting.'));
+        var self = this;
+
+        _.forEach(self.executing, function (irq, id) {
+            irq && clearTimeout(irq);
+            self.emit(`execution.${id}`, new Error(BRIDGE_DISCONNECTING_ERROR_MESSAGE));
```
Comment (shamasis) [inline lib/index.js]: fixed a bug here where `this` was not referring to correct item. Shows that we never disposed sandbox in any tests or code!! ;-)

File: lib/sandbox/legacy-execute.js [hunk 2]
```diff
@@ -57,8 +57,10 @@ module.exports = {
                         if (options && options.globals && options.globals.tests) {
                             for (prop in options.globals.tests) {
                                 // set test to false if its value is undefined.
-                                options.globals.tests.hasOwnProperty(prop) &&
-                                    (options.globals.tests[prop] === undefined) && (options.globals.tests[prop] = false);
+                                if (options.globals.tests.hasOwnProperty(prop)) {
+                                    (options.globals.tests[prop] === undefined) &&
+                                        (options.globals.tests[prop] = false);
+                                }
```
Comment (shamasis) [inline lib/sandbox/legacy-execute.js]: unrelated lint fix

File: package.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     "inherits": "2.0.3",
     "lodash": "4.17.2",
     "uuid": "3.0.1",
-    "uvm": "1.4.0"
+    "uvm": "1.5.0"
```
Comment (shamasis) [inline package.json]: uvm 1.5.0 has the fixes where any dispatch post disconnection does no longer throw things. instead it triggers an error event.
